tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/RabbitMQ/PopulateBasicPropertiesHeadersIntegration.cs

```diff
@@ -45,7 +51,29 @@ public class PopulateBasicPropertiesHeadersIntegration
             return new CallTargetReturn<TReturn?>(returnValue);
         }
 
-        returnValue ??= CachedBasicPropertiesHelper<TReturn>.CreateHeaders();
+        // PopulateBasicPropertiesHeaders returns null if the supplied IReadOnlyBasicProperties
+        // does not have to be modified or if it's a writable instance.
+        // If that is the case then we have to fetch IReadOnlyBasicProperties from the argument
+        // list instead of creating a new instance that overwrites the supplied properties.
+        if (returnValue is null)
+        {
+            // Use the existing basic properties if possible,
+            // if not create new instance using the copy constructor on BasicProperties.
+            if (state.State.DuckIs<IBasicProperties>())
+            {
+                returnValue = (TReturn)state.State!;
+            }
+            else if (state.State.DuckIs<IReadOnlyBasicProperties>())
+            {
+                returnValue = CachedBasicPropertiesHelper<TReturn>.CreateHeaders(state.State!);
```

Comment:
I wonder if we need to preserve / copy headers here?
Haven't tested it out if that is necessary though

Reply:
I'll take a look and see if I can sort it out. We definitely need to copy the basic properties and headers as they will be dropped completely from the message otherwise. For example the current implementation completely breaks our rabbitmq message sender if instrumentation is enabled as the messageid, timestame, persistance settings will be removed.

Reply:
Thank you!
And let us know if need help on this as we definitely want to fix this

Reply:
I have added a commit that should fix the activator, not super happy about the cast but also not sure how to get rid of it. But I suppose optimizations can come after it's working properly :)